apps/wallet-mobile/src/features/SetupWallet/useCases/RestoreWallet/RestoreWalletDetailsScreen.tsx

```diff
@@ -69,12 +69,12 @@ export const RestoreWalletDetailsScreen = () => {
   const bold = useBold()
   const {HEIGHT_MODAL_NAME_PASSWORD, HEIGHT_MODAL_CHECKSUM} = useSizeModal()
   const {openModal, closeModal} = useModal()
-  const walletManager = useWalletManager()
-  const {walletNames} = useWalletNames(walletManager)
+  const {walletManager} = useWalletManager()
+  const walletNames = Array.from(walletManager.walletMetas.values()).map(({name}) => name)
```

Comment:
Would it be interesting to add a method to obtain wallet names?

Reply:
I think it is used only here, it is mainly used to validate if there is any other wallet with same name, maybe wallet-manager should have `hasWalletName(name: string)` ? any thoughts?

Reply:
yap, it looks the logic for renaming is outside the walletManager when it shouldn't since the manager is the responsible to allow/disallow a wallet name, will mark it to refactor.

Reply:
makes sense !!